extensions/libarchive/MergeContent.h

```diff
@@ -259,9 +259,44 @@ class ZipMerge: public ArchiveMerge, public MergeBin {
   }
 };
 
+class AttributeMerger {
+public:
+  AttributeMerger(std::deque<std::shared_ptr<org::apache::nifi::minifi::core::FlowFile>> &flows)
+    : flows_(flows) {}
+  void mergeAttributes(core::ProcessSession *session, std::shared_ptr<core::FlowFile> &mergeFlow);
+  virtual ~AttributeMerger() = default;
+protected:
+  std::map<std::string, std::string> getMergedAttributes();
+  virtual void processFlowFile(const std::shared_ptr<core::FlowFile> &flow, std::map<std::string, std::string>& mergedAttributes) = 0;
+
+  const std::deque<std::shared_ptr<core::FlowFile>> &flows_;
+};
```

Comment:
I decided to go with the classes as the std::accumulate only supports move from c++20 and it would be unoptimal to create a temporal map of every partial result in case of a large number of flow files.